### DISCUSSION TITLE
Apache2 proxy not working as expected

### DISCUSSION THREAD

**Discussion by echeloni2p** [DependencyTrack/dependency-track#3822] [Q&A]
Hello

I tried to setup a apache2 reverse proxy and use only port 443 (and let apache2 handle SSL/TLS).
But I cannot login into our onprem service.

I do run the docker-compose solution, it works in http://url:8080 and backend on http://url:8081

I tried this apache2 setup:
        <Location /api>
                #AllowMethods GET POST OPTIONS
                ProxyPreserveHost On
                ProxyPass  http://localhost:8081/api
                ProxyPassReverse  http://localhost:8081/api
        </Location>

        <Location />
                ProxyPreserveHost On
                ProxyPass  http://localhost:8080/
                ProxyPassReverse  http://localhost:8080/
        </Location>
In docker-compose.yml I set

       - API_BASE_URL=https://dependdany.track.url.tld"

For the ReverseProxy and keep backend running on 8081:8080 and frontend on 8080:8080.
I get to login page, I may enter the user data and I get a 405 error in apache2.
And I got no idea whats wrong.

any ideas so far?

thank you

**Comment by mqu** [comment]
I have a working solution with Apache reverse proxy : 
 - application URL: https://dtrack.some-domain.net/  _(redirect to http://localhost:9040/)_
 - API: https://dtrack.some-domain.net/api/  _(redirect to : http://localhost:9041/)_
 
 here is my Apache configuration  : 

```

 <VirtualHost *:80>
  ServerName dtrack.some-domain.net
  Redirect / https://dtrack.some-domain.net/
  ...
</VirtualHost>

<VirtualHost *:443>

	ServerName dtrack.some-domain.net
	# some stuff about SSL...

	# Configure SSL
	SSLEngine on
	RewriteEngine On

	# API reverse-proxy
	# URL: https://dtrack.some-domain.net/api/
	<Location /api/>
		ProxyPass http://127.0.0.1:9041/api/
		ProxyPassReverse http://127.0.0.1:9041/api/
	</Location>

	# web application frontend
	ProxyPass / http://127.0.0.1:9040/
	ProxyPassReverse / http://127.0.0.1:9040/
	ProxyPreserveHost On

</VirtualHost>
```

- docker-compose.yml part:
```
  app:
    image: dependencytrack/frontend
    depends_on:
      - api
    environment:
      - API_BASE_URL=https://dtrack.some-domain.net
```
 you can test each connexion with CURL:
```
# external side
curl -q --noproxy '*'  https://dtrack.some-domain.net/static/config.json
curl -q --noproxy '*'  https://dtrack.some-domain.net/api/version

# internal side
curl -q --noproxy '*'  http://localhost:9040/static/config.json
curl -q --noproxy '*'  http://localhost:9041/api/version

# now try to login:
curl -d "username=admin&password=dtrack-adm" -X POST http://localhost:9041/api/v1/user/login
curl -d "username=admin&password=dtrack-adm" -X POST https://dtrack.some-domain.net/api/v1/user/login
```
